src/components/Editor.jsx

```diff
@@ -20,6 +35,102 @@ const fetcher = async ( graphQLParams ) => {
 	return response.json();
 };
 
+const PrettifyButton = ( { onClick } ) => {
+	const prettify = usePrettifyEditors();
+	return (
+		<ToolbarButton
+			onClick={ prettify }
+			label="Prettify query (Shift-Ctrl-P)"
+		>
+			<PrettifyIcon
+				className="graphiql-toolbar-icon"
+				aria-hidden="true"
+			/>
+		</ToolbarButton>
+	);
+};
+
+const CopyQueryButton = ( { onCopyQuery } ) => {
+	const copy = useCopyQuery( { onCopyQuery } );
+	return (
+		<ToolbarButton
+			className="graphiql-un-styled graphiql-toolbar-button"
+			onClick={ copy }
+			label="Copy query (Shift-Ctrl-C)"
+		>
+			<CopyIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+		</ToolbarButton>
+	);
+};
+
+const MergeFragmentsButton = () => {
+	const merge = useMergeQuery();
+	return (
+		<ToolbarButton
+			onClick={ merge }
+			label="Merge fragments into query (Shift-Ctrl-M)"
+		>
+			<MergeIcon className="graphiql-toolbar-icon" aria-hidden="true" />
+		</ToolbarButton>
+	);
+};
+
+/**
+ * This button would provide a sharable link to users that they can use to share the current
+ * query and the user that visits the link would see the same query in the editor.
+ *
+ * The way I see this working is that the query would be hashed (similar to current WPGraphQL IDE) but instead of persisting the state in the URL as the query changes,
+ * it would be hashed when the "share" button is clicked and then appended to the current URL or a generic admin URL that could then be shared.
+ *
+ * i.e. site.com/wp-admin/admin.php?page=wpgraphql-ide&query=HASHED_QUERY
+ *
+ * Then GraphiQL would open with the query un-hashed and already set as the current query in the editor.
+ *
+ * This button, or similar buttons to it could/should be able to be "filtered" into GraphiQL by 3rd parties.
+ */
+const ShareDocumentButton = () => {
+	const { dedicatedIdeBaseUrl } = WPGRAPHQL_IDE_DATA;
+	const { queryEditor } = useEditorContext();
+	const [ shareUrl, setShareUrl ] = useState( '' );
+
+	const generateShareLink = () => {
+		const document = queryEditor?.getValue() ?? '';
+		const compressedQuery =
+			LZString.compressToEncodedURIComponent( document );
+		const fullUrl = `${ dedicatedIdeBaseUrl }&query=${ compressedQuery }`;
+
+		setShareUrl( fullUrl );
+		alert( fullUrl ); // Alert with the fullUrl directly since useState is async.
+
+		// Do whatever else we need to here.
+	};
+
+	return (
+		<ToolbarButton
+			onClick={ generateShareLink }
+			label="Share current document"
+		>
+			<Icon
+				icon={ external }
+				style={ {
+					fill: 'hsla(var(--color-neutral), var(--alpha-tertiary))',
```

Comment:
These exact css variables are what the other icons are using. Something to consider abstracting into it's own component for any future theming effort

Reply:
Good callout. 

I included this in my PR for the "help" page / `registerPlugin` PR (#61). . .(although not fully abstracted yet)